src/ExternalSearch.Providers.Dnb/DnBExternalSearchProvider.cs

```diff
@@ -189,24 +209,43 @@ private static IEnumerable<IExternalSearchQueryResult> InternalExecuteSearch(IEx
         throw new ApplicationException("Could not execute external search query - StatusCode:" + cleanseResponse.StatusCode + "; Content: " + cleanseResponse.Content);
     }
 
-    private static string GetAuthToken(DnBExternalSearchJobData jobData)
+
+    private static async Task<string> GetAuthToken(ExecutionContext context, DnBExternalSearchJobData jobData, Guid providerDefinitionId, bool bypassCache)
     {
-        var key = jobData.AuthKey;
-        var secret = jobData.AuthSecret;
-        byte[] bytes = Encoding.ASCII.GetBytes($"{key}:{secret}");
-
-        var restClient = new RestClient(jobData.AuthUrl)
-        {
-            Timeout = -1
-        };
-        var request = new RestRequest(Method.POST);
-        request.AddHeader("Content-Type", "application/json");
-        request.AddHeader("Authorization", $"Basic {Convert.ToBase64String(bytes)}");
-        var body = jobData.AuthRequestBody;
-        request.AddParameter("application/json", body, ParameterType.RequestBody);
-        IRestResponse response = restClient.Execute(request);
-        var responseContent = JsonUtility.Deserialize<AuthResponse>(response.Content);
-        return responseContent.AccessToken;
+        var memoryCache = context.ApplicationContext.Container.Resolve<IMemoryCache>();
+        var cacheKey = $"DnBDirectPlusService.AuthToken({providerDefinitionId})";
+
+        if (!bypassCache && memoryCache.TryGetValue(cacheKey, out var cached))
+        {
+            return (string)cached;
+        }
+
+        using (LockHelper.GetDistributedLockAsync(context.ApplicationContext, $"DnB_GetAuthKey_({providerDefinitionId})_Lock", TimeSpan.FromMinutes(1)).GetAwaiter().GetResult())
```

Comment:
You don't need a distributed lock because you are not using a distributed cache. a simple lock with a static object will do